### ISSUE TITLE
How to make chmod ignore errors?

### ISSUE THREAD

**Issue by sryze** [osxfuse/osxfuse#665]
I would like to make the "chmod" system call not return any errors and fail silently. I need this to make some third-party tools work with files on my NTFS volume. Those tools try to change permissions and fail because chmod returns "Operation not permitted" errors. One example is Android Studio's Gradle plugin.

I think I need to turn on the ["silent" option of NTFS-3G](https://www.tuxera.com/community/open-source-ntfs-3g/) in order to achieve but I'm not sure how to do it with osxfuse. Should I use sysctl for this? How do I find the corresponding sysctl option name?

**Update:**

I added `-o silent` to the mount script in /sbin/mount_ntfs, but nothing changed... any idea why?

```
 13 /usr/local/opt/ntfs-3g/bin/ntfs-3g \
 14   -o volname="${VOLUME_NAME}" \
 15   -o local \
 16   -o negative_vncache \
 17   -o auto_xattr \
 18   -o auto_cache \
 19   -o noatime \
 20   -o windows_names \
 21   -o user_xattr \
 22   -o inherit \
 23   -o uid=$USER_ID \
 24   -o gid=$GROUP_ID \
 25   -o allow_other \
 26   -o silent \
 27   "$@" >> /var/log/mount-ntfs-3g.log 2>&1
```

**Comment by bfleischer** [comment]
NTFS-3G and FUSE for macOS are independent projects. Please report issues related to NTFS-3G to the NTFS-3G maintainers. The previous support forum can be found at http://tuxera.com/forum/viewforum.php?f=7.

**Comment by sryze** [comment]
I'm sorry but the forum that you linked to appears to be dead and I can't register there (it has no registration page). Not to mention their "Mac OS X Questions & Problems" section is locked.

**Comment by sryze** [comment]
I've finally figured this out.

In my /sbin/mount_ntfs script there was a line:

```
USER_ID=502
```

but my actual user ID is 501, so ntfs-3g mounted my NTFS volume with the wrong user ID this whole time! I don't know where 502 came from (GROUP_ID was correct though).

After I replaced the user ID and commented out these lines:

```
if [ `/usr/bin/stat -f %u /dev/console` -ne 0 ]; then
  USER_ID=`/usr/bin/stat -f %u /dev/console`
  GROUP_ID=`/usr/bin/stat -f %g /dev/console`
fi
```

I'm not getting "Operation not permitted" errors anymore 😃
